### ISSUE TITLE
cmake spdlog crash

### ISSUE THREAD

**Issue by StanberyTrask** [muttleyxd/arma3-unix-launcher#176]
hey im trying to recompile because of the recent CDLC and am getting a cmake failure, im on mint 20.1 and the .deb isnt working for whatever reason.
[CMakeError.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7572953/CMakeError.log)
[CMakeOutput.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7572954/CMakeOutput.log)

these are the files cmake says to look at, and on line 101 in CmakeError.log there's
``/home/stanbery/Games/Arma3/arma3-unix-launcher/build/CMakeFiles/CMakeTmp/src.cxx:1:10: fatal error: spdlog/spdlog.h: No such file or directory``

and help would be appreciated.

**Comment by muttleyxd** [comment]
Hi, you would need to install two packages - `libspdlog1 libspdlog-dev`

Command to do it: `sudo apt install libspdlog1 libspdlog-dev`

**Comment by StanberyTrask** [comment]
Thanks for responding, but I have already installed both libspdlog-dev and libspdlog1
```
sudo apt install libspdlog1 libspdlog-dev     
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libspdlog-dev is already the newest version (1:1.5.0-1).
libspdlog1 is already the newest version (1:1.5.0-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```
after confirming this I tried ``cmake ..`` in the build directory again to the same result

**Comment by muttleyxd** [comment]
Please try deleting the build directory and running cmake on clean build/ dir. This should help :+1:

**Comment by StanberyTrask** [comment]
ok, starting with an empty folder i did
```
~/Games/Arma3$ git clone https://github.com/muttleyxd/arma3-unix-launcher.git
Cloning into 'arma3-unix-launcher'...
remote: Enumerating objects: 1790, done.
remote: Counting objects: 100% (322/322), done.
remote: Compressing objects: 100% (220/220), done.
remote: Total 1790 (delta 197), reused 176 (delta 100), pack-reused 1468
Receiving objects: 100% (1790/1790), 1.39 MiB | 4.93 MiB/s, done.
Resolving deltas: 100% (1140/1140), done.
~/Games/Arma3$ cd ./arma3-unix-launcher
~/Games/Arma3/arma3-unix-launcher$ mkdir build && cd build
~/Games/Arma3/arma3-unix-launcher/build$ cmake ..
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test SYSTEM_argparse_WORKS
-- Performing Test SYSTEM_argparse_WORKS - Failed
-- Downloading argparse from https://github.com/p-ranav/argparse.git
-- Using external argparse
-- Performing Test SYSTEM_fmt_WORKS
-- Performing Test SYSTEM_fmt_WORKS - Success
CMake Error at cmake/external_dependencies.cmake:92 (add_library):
  add_library cannot create ALIAS target "fmt::fmt-header-only" because
  target "fmt::fmt" is imported but not globally visible.
Call Stack (most recent call first):
  CMakeLists.txt:26 (setup_fmt)


-- Performing Test SYSTEM_pugixml_WORKS
-- Performing Test SYSTEM_pugixml_WORKS - Success
-- Performing Test SYSTEM_nlohmann_json_WORKS
-- Performing Test SYSTEM_nlohmann_json_WORKS - Failed
-- Performing Test SYSTEM_spdlog_WORKS
-- Performing Test SYSTEM_spdlog_WORKS - Success
-- Configuring incomplete, errors occurred!
See also "~/Games/Arma3/arma3-unix-launcher/build/CMakeFiles/CMakeOutput.log".
See also "~/Games/Arma3/arma3-unix-launcher/build/CMakeFiles/CMakeError.log".
```
these are the logs 
[CMakeError.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7576984/CMakeError.log)
[CMakeOutput.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7576985/CMakeOutput.log)

**Comment by altbrace** [comment]
Same CMake error here. Tried to fix `function(setup_fmt)`, made `fmt::fmt` target global with `set_target_properties(fmt::fmt PROPERTIES IMPORTED_GLOBAL TRUE)`. Build works now, but compilation fails - `<spdlog/cfg/cfg.h>` doesn't exist in the `/usr/include/spdlog` package. Tried building and installing spdlog from source. Still fails. At this point I gave up, will wait for @muttleyxd to figure things out

**Comment by AlbinoJack** [comment]
I just wanted to add that I'm also experiencing what seems to be the same problem with commit 309.

This is the output of cmake ..
```
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test SYSTEM_argparse_WORKS
-- Performing Test SYSTEM_argparse_WORKS - Failed
-- Downloading argparse from https://github.com/p-ranav/argparse.git
-- Using external argparse
-- Performing Test SYSTEM_fmt_WORKS
-- Performing Test SYSTEM_fmt_WORKS - Success
CMake Error at cmake/external_dependencies.cmake:92 (add_library):
  add_library cannot create ALIAS target "fmt::fmt-header-only" because
  target "fmt::fmt" is imported but not globally visible.
Call Stack (most recent call first):
  CMakeLists.txt:26 (setup_fmt)


-- Performing Test SYSTEM_pugixml_WORKS
-- Performing Test SYSTEM_pugixml_WORKS - Failed
-- Downloading pugixml from https://github.com/muttleyxd/pugixml.git
-- Using external pugixml
-- Performing Test SYSTEM_nlohmann_json_WORKS
-- Performing Test SYSTEM_nlohmann_json_WORKS - Failed
-- Performing Test SYSTEM_spdlog_WORKS
-- Performing Test SYSTEM_spdlog_WORKS - Success
-- Configuring incomplete, errors occurred!
See also "/home/[$redacted]/ThirdParty/arma3-unix-launcher/build/CMakeFiles/CMakeOutput.log".
See also "/home/[$redacted]/ThirdParty/arma3-unix-launcher/build/CMakeFiles/CMakeError.log".

```

All requirements to build from source are already installed.

```
cmake is already the newest version (3.16.3-1ubuntu1).
libfmt-dev is already the newest version (6.1.2+ds-2).
libqt5svg5 is already the newest version (5.12.8-0ubuntu1).
libqt5svg5-dev is already the newest version (5.12.8-0ubuntu1).
libqt5widgets5 is already the newest version (5.12.8+dfsg-0ubuntu1).
libspdlog-dev is already the newest version (1:1.5.0-1).
libspdlog1 is already the newest version (1:1.5.0-1).
qt5-default is already the newest version (5.12.8+dfsg-0ubuntu1).

```

[inxi.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7607383/inxi.log)
[CMakeError.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7607388/CMakeError.log)
[CMakeOutput.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7607389/CMakeOutput.log)

**Comment by altbrace** [comment]
I ended up using `.AppImage`. Works well.

**Comment by muttleyxd** [comment]
This should have been fixed by #177 

Feel free to reopen if issue persists
